src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -200,6 +202,8 @@ public Task<SyntaxTree> GetSyntaxTreeAsync(CancellationToken cancellationToken =
 
         /// <summary>
         /// Gets the root node of the current syntax tree if it is available.
+        /// In almost all cases, you should call <see cref="GetSyntaxRootAsync"/> to fetch the root node, which will parse
```

Comment:
If we say in almost all cases we should call the other method then why do we have this method? Should this be changed to "In most cases"?
In general, we have seen several customers hit this issue. Should we add some samples which illustrate the difference?